modules/auxiliary/server/capture/smb.rb

```diff
@@ -25,626 +28,214 @@ def initialize
-      nt_len = ntlm_message.ntlm_response.length
-
-      if nt_len == 24 # lmv1/ntlmv1 or ntlm2_session
-        arg = {
-          :ntlm_ver => NTLM_CONST::NTLM_V1_RESPONSE,
-          :lm_hash => ntlm_message.lm_response.unpack('H*')[0],
-          :nt_hash => ntlm_message.ntlm_response.unpack('H*')[0]
-        }
-
-        if @s_ntlm_esn && arg[:lm_hash][16,32] == '0' * 32
-          arg[:ntlm_ver] = NTLM_CONST::NTLM_2_SESSION_RESPONSE
-        end
-        # if the length of the ntlm response is not 24 then it will be
-        # bigger and represent an NTLMv2 response
-      elsif nt_len > 24 # lmv2/ntlmv2
-        arg = {
-          :ntlm_ver => NTLM_CONST::NTLM_V2_RESPONSE,
-          :lm_hash => ntlm_message.lm_response[0, 16].unpack('H*')[0],
-          :lm_cli_challenge => ntlm_message.lm_response[16, 8].unpack('H*')[0],
-          :nt_hash => ntlm_message.ntlm_response[0, 16].unpack('H*')[0],
-          :nt_cli_challenge => ntlm_message.ntlm_response[16, nt_len - 16].unpack('H*')[0]
-        }
-      elsif nt_len == 0
-        print_status("SMB Capture - Empty hash from #{smb[:name]} - #{smb[:ip]} captured, ignoring ... ")
-        smb_error(CONST::SMB_COM_SESSION_SETUP_ANDX, c, CONST::SMB_STATUS_LOGON_FAILURE, true)
-        return
-      else
-        print_status("SMB Capture - Unknown hash type from #{smb[:name]} - #{smb[:ip]}, ignoring ...")
-        smb_error(CONST::SMB_COM_SESSION_SETUP_ANDX, c, CONST::SMB_STATUS_LOGON_FAILURE, true)
-        return
+      File.open(File.expand_path(datastore['JOHNPWFILE'], Msf::Config.install_root), 'a') do |f|
+        f.puts(combined_hash)
       end
+    end
 
-      buff = pkt['Payload'].v['Payload']
-      buff.slice!(0,securityblobLen)
-      names = buff.split("\x00\x00").map { |x| x.gsub(/\x00/, '') }
-
-      smb[:username] = ntlm_message.user
-      smb[:domain]   = ntlm_message.domain
-      smb[:peer_os]  = names[0]
-      smb[:peer_lm]  = names[1]
-
-      begin
-        smb_get_hash(smb,arg,true)
-      rescue ::Exception => e
-        print_error("SMB Capture - Error processing Hash from #{smb[:name]} - #{smb[:ip]} : #{e.class} #{e} #{e.backtrace}")
+    # Cain & Abel doesn't support import of NTLMv2 hashes
+    if datastore['CAINPWFILE'] && jtr_format == 'netntlm'
+      # Cain&Abel hash format
+      # Username:Domain:Challenge:LMHash:NTLMHash
+      File.open(File.expand_path(datastore['CAINPWFILE'], Msf::Config.install_root), 'a') do |f|
+        f.puts("#{username}:#{domain}:#{server_challenge}:#{client_hash}")
       end
-      smb_error(CONST::SMB_COM_SESSION_SETUP_ANDX, c, CONST::SMB_STATUS_LOGON_FAILURE, true)
-    else
-      smb_error(CONST::SMB_COM_SESSION_SETUP_ANDX, c, CONST::SMB_STATUS_LOGON_FAILURE, true)
     end
   end
 
+  def run
+    @rsock = Rex::Socket::Tcp.create(
+      'LocalHost' => datastore['SRVHOST'],
+      'LocalPort' => datastore['SRVPORT'],
+      'Server' => true,
+      'Timeout' => 3,
```

Comment:
Is this the same as the existing advanced option `SMBServerIdleTimeout`?
```
   SMBServerIdleTimeout    120              yes       The maximum amount of time to keep an idle session open in
                                                      seconds
```

If so should we replace the hardcoded timeout with this option

Reply:
Reading through the code, `Timeout` looks to be the time the socket waits for a request it's expecting, where as `SMBServerIdleTimeout` is how long you let a connection idle _after_ it's been successfully set up.

However we're not even creating a session in this module,  so `SMBServerIdleTimeout` isn't even used. As such, I've deregistered it to avoid confusion.

Reply:
We should replace the hardcoded `3` with a user configurable option at the very least, using an option name which aligns with the rest of the modules :+1: